### ISSUE TITLE
Keeping the deriving clause of a record on the same line as the closing curly brace

### ISSUE THREAD

**Issue by tchoutri** [jaspervdj/stylish-haskell#308]
Hi, I just upgraded to the latest stylish-haskell binary, but I would like to keep the previous behaviour where, in the case of a sole deriving clause, said clause is kept on the same line as the closing curly brace of the record.

Right now, the current settings produce this kind of formatting

```haskell
data Test                  
  = Test { top    :: Int   
         , bottom :: String
         }                 
  deriving (Show)          
````

Which I personally find quite awkward. 

Would you be open to a pull request implementing a configuration option for this behaviour?

**Comment by tchoutri** [comment]
ping @jaspervdj

**Comment by jaspervdj** [comment]
@tchoutri Yep, I don't have much time to do this myself but definitely happy to look at a PR.

However, I'm in the progress of finalizing porting `stylish-haskell` to `ghc-lib-parser` -- see #307.

Would it be possible to branch of `ghc-lib-parser/merge` and PR into that rather than `master`?

**Comment by tchoutri** [comment]
Oh I can even wait for the PR to be completed, no worries about that! Good luck with it, though!

**Comment by jaspervdj** [comment]
@tchoutri The PR is merged now, so feel free to work on this anytime :-)

**Comment by tchoutri** [comment]
@jaspervdj Alright, that's going to be fun then :P

**Comment by tchoutri** [comment]
@jaspervdj I've opened a PR to track my work, but I have to admit some the code in `Step.Data` is a bit nebulous when discovering the codebase. Especially this code:

```haskell
          sep (newline >> spaces cDerivingIndent) $ defn & dd_derivs & \(L pos ds) -> ds <&> \d -> do
              putAllSpanComments (newline >> spaces cDerivingIndent) pos
              putDeriving cfg d
```

Would you be available for a quick chat someday? :)

**Comment by jaspervdj** [comment]
@tchoutri Ah!  Yes, I think that code isn't super idiomatic so if you want to rewrite it or rip it out, feel free.  If it would be any help to you, we can also do so over screenshare, my agenda is pretty open tomorrow afternoon, just drop me an email.

**Comment by jaspervdj** [comment]
@tchoutri How is this progressing?  Let me know if you can use some more help on it. :-)

**Comment by tchoutri** [comment]
@jaspervdj hi! other duties came in the way, but I'll resume my work starting next week. I'll drop you an email if needs be. :)
